lib/rules/no-sibling-hooks.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+module.exports = function (context) {
+    var hooks = [ 'before', 'after', 'beforeEach', 'afterEach' ],
+        isUsed = [];
+
+    function newDescribeLayer(describeNode) {
+        return {
+            describeNode: describeNode,
+            before: false,
+            after: false,
+            beforeEach: false,
+            afterEach: false
+        };
+    }
+
+    return {
+        Program: function (node) {
+            isUsed.push(newDescribeLayer(node));
+        },
+
+        CallExpression: function (node) { // eslint-disable-line complexity, max-statements
+            var name;
+            if (node.callee.type !== 'Identifier') {
+              return;
+            }
+
+            name = node.callee.name;
+            if (name === 'describe') {
```

Comment:
> Should we also support mocha’s TDD interface which uses suite instead of describe?

I don't know how worth it it is as I've never seen `suite` used.
Actually, I'm noticing there are [multiple interfaces](https://mochajs.org/#interfaces) (making me think there should actually be a rule to enforce a consistent interface among the same project).
Do we want to only handle `suite`, or also the other ones described in that section?

Anyway, if we want to support `suite`, we should handle suite(), test(), suiteSetup(), suiteTeardown(), setup(), and teardown() too as long with `before` and co.

I actually don't know what happens when you use both a `before` and a `suiteSetup()` (or is the equivalent `setup()` ?) Does it break or is that valid and should we then consider them to be aliases and report when both are used?

Or maybe we should keep it this way until someone requests it...? :confused:
If the rest of the plugin handles `suite` correctly, then we should, otherwise, it's pretty pointless unless someone wants to invest time to update every concerned rule.

> and maybe also the describe alias context?

Sure. Again, never heard of this one before. There should probably be a rule enforce the use of `before`/`it` or `context`/`specify`. Unless it's so unknown that no one ever uses it.

> Additionally to that should we also support things like xdescribe, xcontext, describe.skip, describe.only etc?

The double use of `before` inside a `describe.only` is still a valid concern, and I think it would then be easy to generalize it to `xdescribe` and friends.